src/Bridge/NelmioApiDoc/Extractor/AnnotationsProvider/ApiPlatformProvider.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Bridge\NelmioApiDoc\Extractor\AnnotationsProvider;
+
+use ApiPlatform\Core\Api\FilterCollection;
+use ApiPlatform\Core\Bridge\NelmioApiDoc\Parser\ApiPlatformParser;
+use ApiPlatform\Core\Bridge\Symfony\Routing\OperationMethodResolverInterface;
+use ApiPlatform\Core\Hydra\ApiDocumentationBuilderInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceNameCollectionFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use Nelmio\ApiDocBundle\Annotation\ApiDoc;
+use Nelmio\ApiDocBundle\Extractor\AnnotationsProviderInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Creates Nelmio ApiDoc annotations for the api platform.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ * @author Teoh Han Hui <teohhanhui@gmail.com>
+ */
+final class ApiPlatformProvider implements AnnotationsProviderInterface
+{
+    private $resourceNameCollectionFactory;
+    private $apiDocumentationBuilder;
+    private $resourceMetadataFactory;
+    private $filters;
+    private $operationMethodResolver;
+
+    public function __construct(ResourceNameCollectionFactoryInterface $resourceNameCollectionFactory, ApiDocumentationBuilderInterface $apiDocumentationBuilder, ResourceMetadataFactoryInterface $resourceMetadataFactory, FilterCollection $filters, OperationMethodResolverInterface $operationMethodResolver)
+    {
+        $this->resourceNameCollectionFactory = $resourceNameCollectionFactory;
+        $this->apiDocumentationBuilder = $apiDocumentationBuilder;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->filters = $filters;
+        $this->operationMethodResolver = $operationMethodResolver;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAnnotations() : array
+    {
+        $annotations = [];
+        $hydraDoc = $this->apiDocumentationBuilder->getApiDocumentation();
+        $entrypointHydraDoc = $this->getResourceHydraDoc($hydraDoc, '#Entrypoint');
+
+        foreach ($this->resourceNameCollectionFactory->create() as $resourceClass) {
+            $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
+
+            $prefixedShortName = ($iri = $resourceMetadata->getIri()) ? $iri : '#'.$resourceMetadata->getShortName();
+            $resourceHydraDoc = $this->getResourceHydraDoc($hydraDoc, $prefixedShortName);
+
+            if ($hydraDoc) {
```

Comment:
@dunglas I'm very curious about this condition. It looks wrong to me... This check would have been done before the loop if this was your intention...

I basically did not change any logic from the previous code. This is the same from https://github.com/nelmio/NelmioApiDocBundle/blob/2a0f95eac0ab2aa466545d0b21d15739636811b7/Extractor/AnnotationsProvider/DunglasApiProvider.php#L71